### ISSUE TITLE
Cannot work with pnpm

### ISSUE THREAD

**Issue by ImSingee** [electric-sql/electric#925]
After upgrading to 0.9.2, can't use with pnpm any more.

Reproduce:

```
npx create-electric-app@latest test
cd test
# change version of electric-sql to 0.9.2
pnpm install
pnpm add zod
pnpm backend:start
pnpm db:migrate
pnpm client:generate
pnpm build
```

Error:

![image](https://uploads.linear.app/0467b297-8a55-4fc0-82da-624730a09e44/7d40c0af-3bf1-441d-9863-4b4f9d9d7587/ca23560a-4d26-4f22-bd0a-69abea3d4367?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzA0NjdiMjk3LThhNTUtNGZjMC04MmRhLTYyNDczMGEwOWU0NC83ZDQwYzBhZi0zYmYxLTQ0MWQtOTg2My00YjRmOWQ5ZDc1ODcvY2EyMzU2MGEtNGQyNi00ZjIyLWJkMGEtNjlhYmVhM2Q0MzY3IiwiaWF0IjoxNzA3MjMyOTY2LCJleHAiOjMzMjc3NzkyOTY2fQ.2v_q3y8MWQBalh3G5RYJgQ-u59D8TdXPBXJSY1bMfG8 "=1724x")

```

> test@0.1.0 build /private/tmp/test2/test
> tsc && vite build

src/generated/client/index.ts:3:38 - error TS6133: 'Relation' is declared but its value is never read.

3 import { type TableSchema, DbSchema, Relation, ElectricClient, type HKT } from 'electric-sql/client/model';
                                       ~~~~~~~~

src/generated/client/index.ts:63:14 - error TS2322: Type 'ZodObject<{ value: ZodOptional<ZodString>; }, "strict", ZodTypeAny, { value?: string | undefined; }, { value?: string | undefined; }>' is not assignable to type 'ZodType<ItemsWhereUniqueInput, ZodTypeDef, ItemsWhereUniqueInput>'.
  Types of property '_type' are incompatible.
    Type '{ value?: string | undefined; }' is not assignable to type 'ItemsWhereUniqueInput'.
      Type '{ value?: string | undefined; }' is not assignable to type '{ value: string; } & { value?: string | undefined; AND?: ItemsWhereInput | ItemsWhereInput[] | undefined; OR?: ItemsWhereInput[] | undefined; NOT?: ItemsWhereInput | ... 1 more ... | undefined; }'.
        Type '{ value?: string | undefined; }' is not assignable to type '{ value: string; }'.
          Types of property 'value' are incompatible.
            Type 'string | undefined' is not assignable to type 'string'.
              Type 'undefined' is not assignable to type 'string'.

63 export const ItemsWhereUniqueInputSchema: z.ZodType<Prisma.ItemsWhereUniqueInput> = z.object({
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: src/generated/client/index.ts:3
```

**Comment by kevin-dp** [comment]
The first issue occurs because our generator does not remove the import for `Relation` if it is not used by the actual schemas. You can fix the issue by removing the import for `Relation` while waiting for us to fix the generator accordingly. I'm not exactly sure why the second issue occurs. Hard to tell without knowing your DB schema. If you can provide your DB schema we can try to reproduce these errors. In any case, these are type errors in the generated client, you can unblock yourself for now by adding a `// ts-ignore` above line 63.

**Comment by ImSingee** [comment]
@kevin-dp The DB is very simple

```
/* This is an example of an SQL DDL migration. It creates an `items` table and
 * then calls an `electric.electrify` procedure to expose the table to the
 * ElectricSQL replication machinery.
 *
 * Note that these statements are applied directly to the *Postgres* database.
 * Electric then handles keeping the local SQLite database schema in sync with
 * the electrified subset of your Postgres database schema.
 *
 * See https://electric-sql.com/docs/usage/data-modelling for more information.
 */

-- Create a simple items table.
CREATE TABLE IF NOT EXISTS items (
  value TEXT PRIMARY KEY NOT NULL
);

-- ⚡
-- Electrify the items table
ALTER TABLE items ENABLE ELECTRIC;
```

(It's just created by create-electric-app)

**Comment by balegas** [comment]
This doesn’t seem to have any particular issue. It should just work. @kevin please try to run commands from 0.9.2

**Comment by fooware** [comment]
The second issue looks like the same problem as in issue #1001, and thus potentially solved by PR #1016.

**Comment by msfstef** [comment]
Closing this in favour of keeping https://github.com/electric-sql/electric/issues/1001 open as they refer to the same issue.

Make sure that your `tsconfig.json` compiler options has `strict` set to `true` (and `strictNullChecks` not set to `false` as it invalidates some of the "strictness" required by Zod, the schema validator).

The `Related` unused import will be fixed by https://github.com/electric-sql/electric/pull/1117.
